File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if(speed==0) return 0;
        if(speed>=1 && speed<=4) return 1;
        if(speed>=5 && speed<=8) return 0.9;
        if(speed==9) return 0.8;
        if(speed==10) return 0.77;
        return 0;
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        double successRte = SuccessRate(speed);
        int carsPerHour = 221;
        return (speed * carsPerHour) * successRte;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        int production = (int) ProductionRatePerHour(speed);
        return production/60;
    }
}

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public int carros = 221;
    public double porcentaje = 1;

    public double productionRatePerHour(int speed) {
        
        if(speed <= 4 && speed > 0){            
            return (speed*carros) * porcentaje;            
        } else if(speed > 4 && speed <=8){            
            porcentaje = 0.9;
            return (speed*carros) * porcentaje;               
        } else if(speed == 9){            
            porcentaje =0.8;
            return (speed*carros) * porcentaje;            
        } else if(speed == 10){            
            porcentaje =0.77;
            return (speed*carros) * porcentaje;            
        } else{
            return 0;
        }
    }

    public int workingItemsPerMinute(int speed) {
        int produccion = (int) productionRatePerHour(speed);
        return produccion / 60;                
    }
}
